library/Haste/Ajax/ReloadHelper.php

```diff
@@ -0,0 +1,235 @@
+    {
+        if (!static::$listeners[$type]) {
+            return $buffer;
+        }
+
+        $id     = (int)$id;
+        $events = [];
+
+        foreach (static::$listeners[$type] as $event => $entries) {
+            if (in_array((int)$id, $entries, true)) {
+                $events[] = $event;
+            }
+        }
+
+        if (count($events) > 0) {
+            $buffer = static::addDataAttributes($buffer, static::getKey($type, $id), $events);
+        }
+
+        return $buffer;
+    }
+
+    /**
+     * Return true if there are listeners
+     *
+     * @return bool
+     */
+    public static function hasListeners()
+    {
+        return count(static::$listeners) > 0;
+    }
+
+    /**
+     * Get the response
+     *
+     * @return JsonResponse|null
+     */
+    public static function getResponse()
+    {
+        if (count(static::$response) < 1) {
+            return null;
+        }
+
+        return new JsonResponse(array_values(static::$response));
+    }
+
+    /**
+     * Return true if the listener is registered
+     *
+     * @param string $type
+     * @param int    $id
+     *
+     * @return bool
+     */
+    public static function isRegistered($type, $id)
+    {
+        if (!static::$listeners[$type]) {
+            return false;
+        }
+
+        $id = (int)$id;
+
+        foreach (static::$listeners[$type] as $entries) {
+            if (in_array($id, $entries, true)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the events
+     *
+     * @param string $type
+     * @param int    $id
+     *
+     * @return array
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function getEvents($type, $id)
+    {
+        if (!static::$listeners[$type]) {
+            throw new \InvalidArgumentException(sprintf('The type "%s" is not subscribed', $type));
+        }
+
+        $events = [];
+        $id     = (int)$id;
+
+        foreach (static::$listeners[$type] as $event => $entries) {
+            if (in_array($id, $entries, true)) {
+                $events[] = $event;
+            }
+        }
+
+        return array_unique($events);
+    }
+
+    /**
+     * Add the data attributes
+     *
+     * @param string $buffer
+     * @param string $id
+     * @param array  $events
+     *
+     * @return string
+     */
+    private static function addDataAttributes($buffer, $id, array $events)
+    {
+        // Merge the data attributes if already present
+        preg_replace_callback(
+            '/\s?data-haste-ajax-id="[^"]*" data-haste-ajax-listeners="([^"]*)"/',
+            function ($matches) use (&$events) {
+                $events = array_merge($events, trimsplit(' ', $matches[1]));
+
+                return '';
+            },
+            $buffer
+        );
+
+        // Remove the HTML comments which break the JS logic
+        $buffer = preg_replace('/<!--(.*)-->/', '', $buffer);
```

Comment:
That's right. The Isotope suffers from the very same issue – you can't reload the product in the reader module if you have the debug mode enabled (which adds the comments).